src/api.ts

```diff
@@ -109,6 +112,10 @@ export const apiFactory = (serviceId: number) => {
 			return res.sendStatus(400);
 		}
 
+		if (workerMap[req.body.common_name] !== req.params.worker) {
```

Comment:
Should the workerMap entry be cleared after we've triggered the disconnect?

Reply:
Certainly can't hurt, done.